demo/transient/transient_snes_mpfaof90.F90

```diff
@@ -217,10 +223,12 @@ program main
     CHKERRA(ierr);
     call DMGetDimension(dm, dim, ierr);
     CHKERRA(ierr);
-    if (dim /= 3) then
-      write(*,*)'Only 3D meshes are supported and the exodus file is not 3D'
-      call exit(0)
-    endif
+    if (dm_plex_extrude_layers > 0) then
+      call DMPlexExtrude(dm, PETSC_DETERMINE, -1.d0, PETSC_TRUE, PETSC_NULL_REAL, PETSC_TRUE, edm, ierr);
```

Comment:
Just curious--the second argument here is `PETSC_DETERMINE`, which means that PETSc itself extracts the number of layers from `-dm_plex_extrude_layers`. Do we still need the call to `PetscOptionsGetInt` at line 181 above, since we don't actually use the value of `dm_plex_extrude_layers`?

Reply:
At the moment, one needs to explicitly call `DMPlexExtrude`, but @knepley is working on it so we can avoid calling it in the future.